src/lib/apis/audio/index.ts

```diff
@@ -12,19 +19,20 @@ export const getAudioConfig = async (token: string) => {
 	})
 		.then(async (res) => {
 			if (!res.ok) throw await res.json();
-			return res.json();
+			const json: Promise<AudioConfigResponse> = res.json();
+			return json;
 		})
 		.catch((err) => {
 			console.log(err);
-			error = err.detail;
+			error = err.detail ?? err;
 			return null;
 		});
 
 	if (error) {
 		throw error;
 	}
 
-	return res;
+	return res!;
```

Comment:
Having to add `!`s like this smells like a bug...

Reply:
TBH the then/catch/err/throw/... async handling here seems like it can be replaced with simple `async`/`await`s unless I'm missing something. See https://github.com/open-webui/open-webui/pull/2227

Reply:
(That said, I'll have to reopen #2227 as a new PR anyway since you can't push to a closed PR, so that doesn't really affect this PR 😅. However, the type-related comments from that PR might apply to this one too.)

Reply:
See: #2233. It might be a good idea to start building types for the API on top of that, if it gets eventually merged?